client/components/isolated-editor/client/components/block-editor-toolbar/header-toolbar/index.js

```diff
@@ -0,0 +1,131 @@
+import { __experimentalToolbarItem as ToolbarItem, Button, Popover } from '@wordpress/components';
+import {
+	BlockToolbar,
+	NavigableToolbar,
+	BlockNavigationDropdown,
+	__experimentalLibrary as Library,
+} from '@wordpress/block-editor';
+import { TableOfContents } from '@wordpress/editor';
+import { plus } from '@wordpress/icons';
+import { useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import EditorHistoryRedo from './redo';
+import EditorHistoryUndo from './undo';
+import './style.scss';
+
+function HeaderToolbar( props ) {
+	const inserterButton = useRef();
+	const { setIsInserterOpened } = useDispatch( 'isolated/editor' );
+	const isMobileViewport = useViewportMatch( 'medium', '<' );
+	const {
+		hasFixedToolbar,
+		isInserterEnabled,
+		isTextModeEnabled,
+		previewDeviceType,
+		isInserterOpened,
+	} = useSelect( ( select ) => {
+		const { hasInserterItems, getBlockRootClientId, getBlockSelectionEnd } = select(
+			'core/block-editor'
+		);
+
+		return {
+			hasFixedToolbar: select( 'isolated/editor' ).isFeatureActive( 'fixedToolbar' ),
+			// This setting (richEditingEnabled) should not live in the block editor's setting.
+			isInserterEnabled:
+				select( 'isolated/editor' ).getEditorMode() === 'visual' &&
+				select( 'core/editor' ).getEditorSettings().richEditingEnabled &&
+				hasInserterItems( getBlockRootClientId( getBlockSelectionEnd() ) ),
+			isTextModeEnabled: select( 'isolated/editor' ).getEditorMode() === 'text',
+			previewDeviceType: 'Desktop',
+			isInserterOpened: select( 'isolated/editor' ).isInserterOpened(),
+		};
+	}, [] );
+	const isLargeViewport = useViewportMatch( 'medium' );
+	const { inserter, toc, navigation, undo } = props.settings.iso.toolbar;
+	const displayBlockToolbar =
+		! isLargeViewport || previewDeviceType !== 'Desktop' || hasFixedToolbar;
+	const toolbarAriaLabel = displayBlockToolbar
+		? /* translators: accessibility text for the editor toolbar when Top Toolbar is on */
+		  __( 'Document and block tools' )
+		: /* translators: accessibility text for the editor toolbar when Top Toolbar is off */
+		  __( 'Document tools' );
+
+	return (
+		<NavigableToolbar className="edit-post-header-toolbar" aria-label={ toolbarAriaLabel }>
+			<div className="edit-post-header-toolbar__left">
+				{ inserter && (
+					<ToolbarItem
+						ref={ inserterButton }
+						as={ Button }
+						className="edit-post-header-toolbar__inserter-toggle"
+						isPrimary
+						isPressed={ isInserterOpened }
+						onMouseDown={ ( event ) => {
+							event.preventDefault();
+						} }
+						onClick={ () => {
+							if ( isInserterOpened ) {
+								// Focusing the inserter button closes the inserter popover
+								inserterButton.current.focus();
+							} else {
+								setIsInserterOpened( true );
+							}
+						} }
+						disabled={ ! isInserterEnabled }
+						icon={ plus }
+						label={ _x( 'Add block', 'Generic label for block inserter button' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **100** times:
`translate( 'Add block' )` *ES Score: 10*


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?